### PR TITLE
Issue 44

### PR DESCRIPTION
**Pull request by jgarciamesa**
Added functionality to specify regions for vcf/bcf files on dng-call with BED files.

### CODE REVIEW

**Comment by reedacartwright** [comment]
Jenkins, ok to test

**Comment by reedacartwright** [comment]
@jgarc111 Does dng-call support the region format for vcf files like it does for bam input?

**Comment by jgarciamesa** [comment]
@reedacartwright Yes, the region format can be either a BED file or a string of the form "chr:beginning-end". 
I forgot to mention that in order to work an indexed vcf/bcf file must exist on the same folder. 
I will create a new test case with --region option for dng call.

**Comment by jgarciamesa** [comment]
@reedacartwright We currently are not able to specify regions with .BED files for bam input. Do you want me to implement that or is there anything else with higher priority right now?

**Comment by reedacartwright** [comment]
@jgarc111 Yes, we should support both our region format and bed format for both bam and bcf formats.

**Comment by jgarciamesa** [comment]
Now bed format is supported for both bcf and bam files on dng call. In order to work, the bcf and bam files must be sorted and there must exist on the same directory their index file. For bcf, bcftools can be used to generate those files. For bam, samtools can be used to generate them.

**Comment by reedacartwright** [comment]
Jenkins, please retest

**Comment by reedacartwright** [comment]
Jenkins, retest this please

File: src/lib/call.cc [hunk 1]
```diff
@@ -313,12 +315,35 @@ int task::Call::operator()(Call::argument_type &arg) {
...
+				getline(inFile, line);
+				//skip header lines
+				if((line.find("browser")!=std::string::npos) || (line.find("track")!=std::string::npos)) continue;
+				//convert bed information into "chr:start-end" format
+				line.replace(line.find('\t'),1,":");
+				line.replace(line.find('\t'),1,"-");
+				//only keep columns containing chromosome name, start and end position
+				if(line.find('\t')!=std::string::npos) line.erase(line.find('\t'),line.size());
+				regs += " " + line;
+			}
```
Comment (reedacartwright) [inline src/lib/call.cc]: BED is 0-based and chr:start-end is 1-based. This code does not consider that.

**Comment by reedacartwright** [comment]
Instead of creating a string from the BED file, considering putting code int `src/lib/regions.cc` to support `bed` files.  `dng::regions::bam_parse` parses an region string based on the contigs contained in a bam file.  You can resuse a lot of code and split that function into `dng::regions::bam_parse_region` and `dng::regions::bam_parse_bed`.

**Comment by reedacartwright** [comment]
Where are with with these revisions?

**Comment by jgarciamesa** [comment]
I expect to get it done by the end of this week. I'm sorry last week I couldn't get enough work done because of my final exams.

**Comment by reedacartwright** [comment]
Fixed in 614f08c6e1

**Pull request closed without merging**